crates/matrix-sdk/tests/integration/widget.rs

```diff
@@ -310,9 +325,108 @@ async fn read_room_members() {
     }
 }
 
+#[async_test]
+async fn receive_live_events() {
+    let (client, mock_server, driver_handle) = run_test_driver(false).await;
+
+    negotiate_capabilities(
+        &driver_handle,
+        json!([
+            "org.matrix.msc2762.receive.event:m.room.member",
+            "org.matrix.msc2762.receive.event:m.room.message#m.text",
+            "org.matrix.msc2762.receive.state_event:m.room.name#",
+            "org.matrix.msc2762.receive.state_event:m.room.member#@example:localhost",
+        ]),
+    )
+    .await;
+
+    // No messages from the driver yet
+    assert_matches!(recv_message(&driver_handle).now_or_never(), None);
+
+    let mut sync_builder = SyncResponseBuilder::new();
+    // bump the internal batch counter, otherwise the response will be seen as
+    // identical to the one done in `run_test_driver`
+    sync_builder.build_json_sync_response();
```

Comment:
There's also a call to `build_json_sync_response` in `run_test_driver`, shouldn't that bump the internal batch counter?

Reply:
It's a different `SyncResponseBuilder` instance. I didn't want to make that function return *yet* another object.

Reply:
Maybe the function could take a `&mut SyncResponseBuilder` parameter instead? Or return a struct containing all the returned fields at once, so adding a new one doesn't require modifying all the callers?

Reply:
Yeah I think this can be improved, but I'm not sure what's really the best option. The `&mut` parameter would be annoying for all the other tests.